### PR TITLE
Add lints

### PR DESCRIPTION
**Pull request by breeswish**
This PR adds lints to routine development as well as CI check.

You can build with lint using the same way as other TiDB projects:

```sh
make dev
```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+run:
+  sip-dirs:
```
Comment (HunDunDM) [inline .golangci.yml]: skip-dirs?

File: Makefile [hunk 2]
```diff
@@ -24,8 +24,10 @@ LDFLAGS += -X "$(DASHBOARD_PKG)/pkg/utils.GitBranch=$(shell git rev-parse --abbr
 default:
 	SWAGGER=1 make server
 
-tidy:
-	go mod tidy
+lint:
```
Comment (HunDunDM) [inline Makefile]: Need swagger_spec as a target?

Comment (breeswish) [reply Makefile]: this may make it impossible to lint without swagger (swagger dependencies are always installed)?

Comment (HunDunDM) [reply Makefile]: Sorry, I didn't see `docs/placeholder.go` before

**Pull request merged**